src/hooks/useAssignCard.ts

```diff
@@ -0,0 +1,179 @@
+    const [currencyList = getEmptyObject<CurrencyList>()] = useOnyx(ONYXKEYS.CURRENCY_LIST, {canBeMissing: true});
+
+    const filteredCardList = getFilteredCardList(cardsList, selectedFeed ? cardFeeds?.[selectedFeed]?.accountList : undefined, workspaceCardFeeds);
+
+    const companyCards = getCompanyFeeds(cardFeeds);
+    const selectedFeedData = selectedFeed && companyCards[selectedFeed];
+    const domainOrWorkspaceAccountID = getDomainOrWorkspaceAccountID(workspaceAccountID, selectedFeedData);
+
+    const fetchCompanyCards = () => {
+        openPolicyCompanyCardsPage(policyID, domainOrWorkspaceAccountID);
+    };
+
+    const {isOffline} = useNetwork({onReconnect: fetchCompanyCards});
+
+    const filteredFeedCards = filterInactiveCards(allFeedsCards?.[`${ONYXKEYS.COLLECTION.WORKSPACE_CARDS_LIST}${domainOrWorkspaceAccountID}_${selectedFeed}`]);
+    const hasFeedError = selectedFeed ? !!cardFeeds?.[selectedFeed]?.errors : false;
+    const isSelectedFeedConnectionBroken = checkIfFeedConnectionIsBroken(filteredFeedCards) || hasFeedError;
+    const isAllowedToIssueCompanyCard = useIsAllowedToIssueCompanyCard({policyID});
+
+    const {isActingAsDelegate, showDelegateNoAccessModal} = useContext(DelegateNoAccessContext);
+
+    const isAssigningCardDisabled = !currentFeedData || !!currentFeedData?.pending || isSelectedFeedConnectionBroken || !isAllowedToIssueCompanyCard;
+
+    const assignCard = (cardID: string) => {
+        if (isAssigningCardDisabled) {
+            return;
+        }
+
+        if (isActingAsDelegate) {
+            showDelegateNoAccessModal();
+            return;
+        }
+
+        if (!selectedFeed) {
+            return;
+        }
+
+        const isCommercialFeed = isCustomFeed(selectedFeed);
+
+        // If the feed is a direct feed (not a commercial feed) and the user is offline,
+        // show the offline alert modal to inform them of the connectivity issue.
+        if (!isCommercialFeed && isOffline) {
+            setShouldShowOfflineModal(true);
+            return;
+        }
+
+        const data: Partial<AssignCardData> = {
+            bankName: feed,
+        };
+
+        if (cardID) {
+            data.encryptedCardNumber = cardID;
+        }
+
+        let currentStep: AssignCardStep = CONST.COMPANY_CARD.STEP.ASSIGNEE;
+        const employeeList = Object.values(policy?.employeeList ?? {}).filter((employee) => !isDeletedPolicyEmployee(employee, isOffline));
+        const isFeedExpired = isSelectedFeedExpired(selectedFeedData);
+        const plaidAccessToken = selectedFeedData?.plaidAccessToken;
+
+        // Refetch plaid card list
+        if (!isFeedExpired && plaidAccessToken) {
+            const country = selectedFeedData?.country ?? '';
+            importPlaidAccounts('', selectedFeed, '', country, getDomainNameForPolicy(policyID), '', undefined, undefined, plaidAccessToken);
+        }
+
+        if (!cardID && employeeList.length === 1) {
```

Comment:
Each `if` statement could use a comment to explain why that logic is necessary. It's not very clear to me why we need to do all the logic.